tests/testthat/test-test_basic_finalsize.R

```diff
@@ -21,3 +23,72 @@ test_that("Check basic final size calculation works", {
     length(p_suscep), length(epi_final_size)
   )
 })
+
+test_that("Check failure for unequal contact matrix and demography", {
+  polymod <- socialmixr::polymod
+  contact_data <- socialmixr::contact_matrix(
+    polymod,
+    countries = "United Kingdom",
+    age.limits = c(0, 20, 40)
+  )
+  c_matrix <- t(contact_data$matrix)
+  d_vector <- c(0.5, 0.5) # manually created
+  p_suscep <- c(1, 1, 1)
+  p_initial_infections <- c(0.1, 0.5, 0.5)
+
+  testthat::expect_error(
+    final_size(
+      r0 = 2,
+      contact_matrix = c_matrix,
+      demography_vector = d_vector,
+      prop_initial_infected = p_initial_infections,
+      prop_suscep = p_suscep
+    )
+  )
```

Comment:
Here and below, it would be useful / more robust to specify the expected character string for the error/warning/message (second argument of `expect_error`). Something along the lines of:

```
msg <- "The error message that should be generated"
expect_error(<code generating error>, msg)

```

Reply:
Fair, now checking for string content using regexp, added in https://github.com/epiverse-trace/finalsize/pull/19/commits/654b47e13f76c12077360980baac028e5eea09c4